internal/app/mapper/mapper.go

```diff
@@ -34,5 +38,44 @@ func (mapper *mapper) GetKey(request v2.DiscoveryRequest, typeURL string) (strin
 	if typeURL == "" {
 		return "", fmt.Errorf("typeURL is empty")
 	}
+
+	var resultFragments []string
+	for _, fragment := range mapper.config.GetFragments() {
+		fragmentrules := fragment.GetRules()
+		for _, fragmentrule := range fragmentrules {
+			matchpredicate := fragmentrule.GetMatch()
+			isMatch := isMatchPredicate(matchpredicate)
+			if isMatch {
+				result := getResult(fragmentrule)
+				resultFragments = append(resultFragments, result)
+			}
+		}
+	}
+
+	if len(resultFragments) != 0 {
+		return strings.Join(resultFragments, "_"), nil
+	}
+
 	return "", fmt.Errorf("Cannot map the input to a key")
 }
+
+func isMatchPredicate(matchPredicate *matchPredicate) bool {
```

Comment:
shouldn't this be `isAnyMatch(...)`?
if it's to support the other future types would prefer to name this `isMatch(...)`

Reply:
More predicates are coming the next PR in this function. `isMatch` sounds good